### PR TITLE
Test plugin on folders

### PR DESCRIPTION
**Pull request by jdmota**
Based on your [comment](https://github.com/amilajack/eslint-plugin-flowtype-errors/pull/75#issuecomment-289231942), I decided to try to implement a way to test the plugin on different folders.

I used the snapshot feature from Jest. To update the snapshots, one can just run `npm run spec -- -u`.

I think this also makes it easier for other people to add more tests.

Tell me if you like this solution.

### CODE REVIEW

File: test/format.spec.js [hunk 1]
```diff
@@ -54,3 +55,36 @@ describe('Format', () => {
...
+      const { stdout, stderr } = runEslint(fullFolder);
+
+      const regexp = new RegExp(`^${fullFolder.replace(/[\\^$.*+?()[\]{}|]/g, '\\$&')}.+\\.js$`, 'gm'); // Escape regexp
+
+      // Strip root from filenames
+      expect(stdout.replace(regexp, match => match.replace(fullFolder, '.').replace(/\\/g, '/'))).toMatchSnapshot();
+
+      expect(stderr).toEqual('');
+
+      done();
```
Comment (amilajack) [inline test/format.spec.js]: We use jest, which doesn't require using done for tests. Try removing the `done()` and see if the tests still behave as expected

**Comment by amilajack** [comment]
> Tell me if you like this solution.

Love it! Would be great if we could use link the package after building it and use that version instead

Maybe something like
```bash
npm run build
npm link
cd test-project-1
npm link eslint-plugin-flowtype-errors
npm run spec
```

**Comment by jdmota** [comment]
I thought of that. But it seems that `npm link` creates a global link. Not sure if that can create conflicts if, for example, I'm using eslint-plugin-flowtype-errors in another project...

The hack I used should be faster than running `npm` commands I think, although using those would avoid the need for having a `.eslintrc.js` file in each folder...

**Comment by amilajack** [comment]
Maybe we should add those scripts to travis and appveyor so they only run in CI

**Comment by jdmota** [comment]
Running this tests, I think, it's useful during development, and not just for CI. And you also need to be able to generate the snapshots.

Just pushed a new commit to create .eslintrc.js files dynamically.

**Comment by amilajack** [comment]
Works like a charm. Thanks again!

**Comment by jdmota** [comment]
You're welcome!

I wonder if with this, we still need #52 ?

**Comment by amilajack** [comment]
Nope! Just closed it.

**Pull request merged**